### DISCUSSION TITLE
[Q&A] How to unfocus fields after submit form

### DISCUSSION THREAD

**Discussion by nhattran998** [react-hook-form/react-hook-form#6308] [Q&A] [answered]
After submitting the form, I want to unfocus a field or multiple fields, so hope anyone can help me with that <3. Thanks

**Comment by bluebill1049** [comment]
read the doc under useForm

**Comment by kaisiant** [comment]
I also having same this issue.

1.  2 input fields with "firstName" and "lastName", both are required fields
2.  input value into "firstName" and press Enter
3. "lastName" get focused, input value and press Enter
4. form get submitted but the focus remains at "lastName" (not sure is this the web behaviour)

What I can think of is using a hidden input and after submitted the form setFocus to that hidden input.

Any help would be great. Thanks.

**Comment by Moshyfawn** [comment]
This is not an issue and is nothing to do with RHF but with the native browser behavior. Try removing RHF from your form: it's gonna emit the same behavior not unfocusing your focused element on submitting a form via `Enter` key.

I can advise to do something like the code snippet below `onSubmit` to unfocus the currently focused element.
```js
document.activeElement?.blur()
```

**Comment by ritikbanger** [comment, marked as answer]
Two ways to do the same.

1. Blur the focused element: `document.activeElement?.blur()`

2. Add the following HTML somewhere on your page:
`<input id="blur-hack" type="text" style="position: absolute; opacity: 0;">`
Then this JS will unfocus anything currently focused:
`document.getElementById("blur-hack").focus();`

**Reply by Moshyfawn** [reply]
> 2. Add the following HTML somewhere on your page:
`<input id="blur-hack" type="text" style="position: absolute; opacity: 0;">`
Then this JS will unfocus anything currently focused:
`document.getElementById("blur-hack").focus();`

You can just provide a `type="hidden"` input instead of doing CSS in your second example.

Also, the OP has already thought of hidden input approach
> What I can think of is using a hidden input and after submitted the form setFocus to that hidden input.

and [I've already mentioned](https://github.com/react-hook-form/react-hook-form/discussions/6308#discussioncomment-2484075)
```js
document.activeElement?.blur()
```
